server/src/main/java/org/opensearch/action/admin/cluster/decommission/awareness/get/GetDecommissionStateResponse.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.decommission.awareness.get;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.action.ActionResponse;
+import org.opensearch.cluster.decommission.DecommissionAttribute;
+import org.opensearch.cluster.decommission.DecommissionStatus;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Objects;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+/**
+ * Response for decommission status
+ *
+ * @opensearch.internal
+ */
+public class GetDecommissionStateResponse extends ActionResponse implements ToXContentObject {
+
+    private final DecommissionAttribute decommissionedAttribute;
+    private final DecommissionStatus status;
+
+    GetDecommissionStateResponse() {
+        this(null, null);
+    }
+
+    GetDecommissionStateResponse(DecommissionAttribute decommissionedAttribute, DecommissionStatus status) {
+        this.decommissionedAttribute = decommissionedAttribute;
+        this.status = status;
+    }
+
+    GetDecommissionStateResponse(StreamInput in) throws IOException {
+        if (in.readBoolean()) {
+            this.decommissionedAttribute = new DecommissionAttribute(in);
+        } else {
+            this.decommissionedAttribute = null;
+        }
+        if (in.readBoolean()) {
+            this.status = DecommissionStatus.fromString(in.readString());
+        } else {
+            this.status = null;
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        boolean isNotNullDecommissionAttribute = this.decommissionedAttribute != null;
+        boolean isNotNullStatus = this.status != null;
+        out.writeBoolean(isNotNullDecommissionAttribute);
+        if (isNotNullDecommissionAttribute) {
+            decommissionedAttribute.writeTo(out);
+        }
+        out.writeBoolean(isNotNullStatus);
+        if (isNotNullStatus) {
+            out.writeString(status.status());
+        }
+    }
```

Comment:
Yup, tried that earlier, but that would lead to unnecessary null pointer handling for `status.status()` and else where as well. Also, writing a boolean for checking `null` and handling it is used at multiple places for such use cases in OpenSearch and hence preferred this approach.. looks cleaner as well